tests/lib/ocpp/v201/test_smart_charging_handler.cpp

```diff
@@ -170,64 +176,122 @@ TEST_F(ChargepointTestFixtureV201, K01FR09_IfTxProfileEvseHasNoActiveTransaction
+    auto sut = handler.validate_tx_profile(profile_1, *evses[DEFAULT_EVSE_ID]);
 
     EXPECT_THAT(sut, testing::Eq(ProfileValidationResultEnum::Valid));
 }
 
-} // namespace ocpp::v201
+
+/*
+ * 0 - For K01.FR.52, we return DuplicateTxDefaultProfileFound if an existing profile is on an EVSE, 
+ *  and a new profile with the same stack level and different profile ID will be associated with EVSE ID = 0.
+ * 1 - For K01.FR.53, we return DuplicateTxDefaultProfileFound if an existing profile is on an EVSE ID = 0, 
+ *  and a new profile with the same stack level and different profile ID will be associated with an EVSE.
+ * 2-7 - We return Valid for any other case, such as using the same EVSE or using the same profile ID. 
+ */
+
+INSTANTIATE_TEST_SUITE_P(TxDefaultProfileValidationV201_Param_Test_Instantiate,
+                            ChargepointTestFixtureV201,
+                            testing::Values(
+                                std::make_tuple(DEFAULT_EVSE_ID, STATION_WIDE_ID, DEFAULT_PROFILE_ID + 1, DEFAULT_STACK_LEVEL, ProfileValidationResultEnum::DuplicateTxDefaultProfileFound), 
+                                std::make_tuple(STATION_WIDE_ID, DEFAULT_EVSE_ID, DEFAULT_PROFILE_ID + 1, DEFAULT_STACK_LEVEL, ProfileValidationResultEnum::DuplicateTxDefaultProfileFound),
+                                std::make_tuple(STATION_WIDE_ID, STATION_WIDE_ID, DEFAULT_PROFILE_ID + 1, DEFAULT_STACK_LEVEL, ProfileValidationResultEnum::Valid),
+                                std::make_tuple(DEFAULT_EVSE_ID, DEFAULT_EVSE_ID, DEFAULT_PROFILE_ID + 1, DEFAULT_STACK_LEVEL, ProfileValidationResultEnum::Valid),
+                                std::make_tuple(DEFAULT_EVSE_ID, STATION_WIDE_ID, DEFAULT_PROFILE_ID, DEFAULT_STACK_LEVEL, ProfileValidationResultEnum::Valid),
+                                std::make_tuple(STATION_WIDE_ID, DEFAULT_EVSE_ID, DEFAULT_PROFILE_ID, DEFAULT_STACK_LEVEL, ProfileValidationResultEnum::Valid),
+                                std::make_tuple(DEFAULT_EVSE_ID, STATION_WIDE_ID, DEFAULT_PROFILE_ID + 1, DEFAULT_STACK_LEVEL + 1, ProfileValidationResultEnum::Valid), 
+                                std::make_tuple(STATION_WIDE_ID, DEFAULT_EVSE_ID, DEFAULT_PROFILE_ID + 1, DEFAULT_STACK_LEVEL + 1, ProfileValidationResultEnum::Valid)
+                            ));
+
+TEST_P(ChargepointTestFixtureV201, K01FR52_and_K01FR53_TxDefaultProfileValidationV201Tests) {
+    auto [existing_evse_id, added_evse_id, added_profile_id, added_stack_level, expected] = GetParam();
+    install_profile_on_evse(existing_evse_id, DEFAULT_PROFILE_ID);
+
+    create_evse_with_id(added_evse_id);
+    auto profile = create_charging_profile(added_profile_id, ChargingProfilePurposeEnum::TxDefaultProfile, create_charge_schedule(ChargingRateUnitEnum::A), uuid(), added_stack_level);
+    auto sut = handler.validate_tx_default_profile(profile, *evses[added_evse_id]);
+
+    EXPECT_THAT(sut, testing::Eq(expected));
+}
+
+// TODO: delete me
```

Comment:
NOTE: The above parameterized test replaces the following two tests (which only cover the first two parameterized values).

Reply:
To clarify, the below tests will get removed since they are covered by the parameterized tests above. I just wanted to show them in this draft PR.